jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/DataTableAssert.java

```diff
@@ -27,154 +27,179 @@
+    public DataTableAssert<L, D> value(JFunc1<D, Boolean> condition, Row row, String... messages) {
+        jdiAssert(condition.execute(table().dataRow(row.getIndex(table().header()))), Matchers.is(true), messages);
         return this;
     }
+    
     /**
      * Check that the table has row that has expected data
+     *
      * @param data to compare
      * @return DataTableAssert
      */
     @JDIAction("Assert that '{name}' has {0}")
-    public DataTableAssert<L, D> row(D data) {
-        return row(d -> d.equals(data));
+    public DataTableAssert<L, D> row(D data, String ... messages) {
+        return row(d -> d.equals(data), messages);
     }
+    
     @JDIAction("Assert that '{name}' has {0}")
-    public DataTableAssert<L, D> value(D expected, Row row) {
+    public DataTableAssert<L, D> value(D expected, Row row, String ... messages) {
         D actual = table().dataRow(row.getIndex(table().header()));
-        jdiAssert(actual, Matchers.is(expected));
+        jdiAssert(actual, Matchers.is(expected), messages);
         return this;
     }
+    
     @JDIAction("Assert that '{name}' rows {0}")
-    public DataTableAssert<L, D> rows(Matcher<? super List<D>> condition) {
-        jdiAssert(table().allData(), condition);
+    public DataTableAssert<L, D> rows(Matcher<? super List<D>> condition, String ... messages) {
+        jdiAssert(table().allData(), condition, messages);
         return this;
     }
+    
     @JDIAction("Assert that '{name}' is sorted")
-    public DataTableAssert<L, D> sortedBy(JFunc2<L, L,Boolean> condition) {
+    public DataTableAssert<L, D> sortedBy(JFunc2<L, L, Boolean> condition, String ... messages) {
         List<L> allRows = table().allLines();
         for (int i = 1; i < allRows.size(); i++)
-            if (!condition.execute(allRows.get(i-1), allRows.get(i)))
-                jdiAssert("Table not sorted at "+i+" row", Matchers.is(""));
-        jdiAssert("Table is sorted", Matchers.is("Table is sorted"));
+            if (!condition.execute(allRows.get(i - 1), allRows.get(i)))
+                jdiAssert("Table not sorted at " + i + " row", Matchers.is(""), messages);
+        jdiAssert("Table is sorted", Matchers.is("Table is sorted"), messages);
         return this;
     }
+    
     @JDIAction("Assert that '{name}' is sorted")
-    public DataTableAssert<L, D> dataSortedBy(JFunc2<D, D,Boolean> condition) {
+    public DataTableAssert<L, D> dataSortedBy(JFunc2<D, D, Boolean> condition, String ... messages) {
         List<D> allRows = table().allData();
         for (int i = 1; i < allRows.size(); i++)
-            if (!condition.execute(allRows.get(i-1), allRows.get(i)))
-                jdiAssert("Table not sorted at "+i+" row", Matchers.is(""));
-        jdiAssert("Table is sorted", Matchers.is("Table is sorted"));
+            if (!condition.execute(allRows.get(i - 1), allRows.get(i)))
+                jdiAssert("Table not sorted at " + i + " row", Matchers.is(""), messages);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5133577)